platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java

```diff
@@ -277,18 +259,31 @@ public void putValue(final int keyHashCode, final K key, final long value) {
             /* We have to serialize a variable-size key to a temp byte buffer to check
             if there is going to be enough room to store it in this bucket. */
             if (keySerializer.isVariableSize()) {
-                reusableDataFileOutputStream.reset();
-                keySerializer.serialize(key, reusableDataFileOutputStream);
-                reusableDataFileOutputStream.flush();
-                final int keySizeBytes = reusableDataFileOutputStream.bytesWritten();
+                reusableBuffer.rewind();
+                while (true) {
+                    try {
+                        keySerializer.serialize(key, reusableBuffer);
```

Comment:
Is there any way to get the size (even if just a sort-of-close estimate) of these without serializing them?  This loop might get pretty costly for large objects, particularly if serialization is expensive and/or if the object is particularly large (due to repeatedly allocating new buffers).

Reply:
This loop (which doesn't look nice, I agree) is when keys are of variable size, no way to predict how large buffer would be enough to host all key bytes. Serialization is not free, but looks pretty cheap, as far as I can see in HS code, it's usually just to put a few bytes + check the number of non-zero bytes in a long.